src/index.ts

```diff
@@ -1,19 +1,649 @@
-// functions
-export * from "./lib/mapping/connect";
-export * from "./lib/mapping/decrypt";
-export * from "./lib/mapping/disconnect";
-export * from "./lib/mapping/encrypt";
-export * from "./lib/mapping/getActiveAddress";
-export * from "./lib/mapping/getActivePublicKey";
-export * from "./lib/mapping/getWalletNames";
-export * from "./lib/mapping/sign";
-export * from "./lib/mapping/signature";
-export * from "./lib/mapping/dispatch";
-export * from "./lib/mapping/signMessage";
-export * from "./lib/mapping/verifyMessage";
-export * from "./lib/mapping/signDataItem";
-// types
-export * from "./types/mapping/connect";
-
 import { Buffer } from "buffer";
+import { OthentAuth0Client } from "./lib/auth/auth0";
+import Transaction, { Tag } from "arweave/web/lib/transaction";
+import { addOthentAnalyticsTags } from "./lib/analytics/analytics.utils";
+import type Arweave from "arweave/web";
+import {
+  BinaryDataType,
+  binaryDataTypeOrStringToBinaryDataType,
+  bufferTob64Url,
+  hash,
+  stringToUint8Array,
+} from "./lib/utils/arweaveUtils";
+import { createData, DataItemCreateOptions, Signer } from "warp-arbundles";
+import { OthentKMSClient } from "./lib/othent-kms-client/client";
+import {
+  Auth0Strategy,
+  UserDetails,
+} from "./lib/auth/auth0.types";
+import {
+  ArConnect,
+  DataItem,
+  DispatchResult,
+  PermissionType,
+  SignMessageOptions,
+} from "./types/arconnect/arconnect.types";
+import {
+  ANALYTICS_TAGS,
+  CLIENT_NAME,
+  CLIENT_VERSION,
+  DEFAULT_OTHENT_CONFIG,
+} from "./lib/config/config.constants";
+
+// Type exports:
+
+export {
+  DispatchResult,
+  SignMessageOptions,
+} from "./types/arconnect/arconnect.types";
+export { TypedArray, BinaryDataType } from "./lib/utils/arweaveUtils";
+
+export {
+  Auth0Strategy,
+  IdTokenWithData,
+  UserDetails,
+} from "./lib/auth/auth0.types";
+
+// Constant exports:
+
+export {
+  DEFAULT_OTHENT_CONFIG,
+  CLIENT_NAME,
+  CLIENT_VERSION,
+} from "./lib/config/config.constants";
+
+// TODO: Polyfill. Should we overwrite a global like this from a library?
+
 window.Buffer = Buffer;
```

Comment:
@7i7o Not sure about this, as this will affect other libraries and user land.

Reply:
This was added as a quick patch so the library would work in a framework like Vite where Buffer (and many other packages) are not provided by default.
If you find a more elegant solution, that would be very well welcomed

Reply:
Ok, for now I've just created an issue to keep track of this: https://github.com/Othent/KeyManagementService/issues/25

I would probably check if `Buffer` already exists before polyfilling it.